infra/cifuzz/run_fuzzers.py

```diff
@@ -202,6 +209,55 @@ class BatchFuzzTargetRunner(BaseFuzzTargetRunner):
   def quit_on_bug_found(self):
     return False
 
+  def run_fuzz_target(self, fuzz_target_obj):
+    """Fuzzes with |fuzz_target_obj| and returns the result."""
+    result = fuzz_target_obj.fuzz()
+    logging.debug('corpus_path: %s', os.listdir(result.corpus_path))
+    self.clusterfuzz_deployment.upload_corpus(fuzz_target_obj.target_name,
+                                              result.corpus_path)
+    # This must be done after we upload the corpus, otherwise it will be deleted
+    # before we get a chance to upload it. We can't delete the fuzz target
+    # because it is needed when we upload the build.
+    fuzz_target_obj.free_disk_if_needed(delete_fuzz_target=False)
+    return result
+
+  def run_fuzz_targets(self):
+    result = super().run_fuzz_targets()
+
+    self.clusterfuzz_deployment.upload_crashes(self.crashes_dir)
+
+    # We want to upload the build to the filestore after we do batch fuzzing.
+    # There are some problems with this. First, we don't want to upload the
+    # build before fuzzing, because if we download the latest build, we will
+    # consider the build we just uploaded to be the latest even though it
```

Comment:
can you explain this a bit more? is this because we need the "latest" build to be the previous one for novel crash detection?

Reply:
Duplicate comment